pandas/tests/indexes/test_base.py

```diff
@@ -1615,6 +1632,122 @@ def test_string_index_repr(self):
 
                 self.assertEqual(coerce(idx), expected)
 
+    def test_mixed_int_index(self):
```

Comment:
Added these tests. 
(It turned out to be a bit lengthy. maybe split it or/and put in a separate class?)